src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs

```diff
@@ -165,6 +151,22 @@ public CompilationResult Compile(RelativeFileInfo fileInfo, string compilationCo
             }
         }
 
+        /// <summary>
+        /// Gets the <see cref="DependencyContext"/>.
+        /// </summary>
+        /// <param name="hostingEnvironment">The <see cref="IHostingEnvironment"/>.</param>
+        /// <returns>The <see cref="DependencyContext"/>.</returns>
+        protected virtual DependencyContext GetDependencyContext(IHostingEnvironment hostingEnvironment)
+        {
+            if (hostingEnvironment.ApplicationName != null)
```

Comment:
It's sometimes null in unit test scenarios. Although I could remove this check now that we only create it when we compile and not as part of the ctor.